### PR TITLE
Move nftables state tests to pytest

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Move nftables state tests to pytest

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/unit/states/test_nftables.py [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+                side_effect=[
+                    {"result": True, "comment": ""},
+                    {"result": False, "comment": ""},
+                    {"result": False, "comment": ""},
+                    {"result": False, "comment": ""},
+                ]
+            )
+            with patch.dict(nftables.__salt__, {"nftables.check": mock}):
+                ret.update(
+                    {"comment": "nftables rule for salt" " already set (a) for ipv4"}
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:110]:
```suggestion
                    {"comment": "nftables rule for salt already set (a) for ipv4"}
```

File: tests/pytests/unit/states/test_nftables.py [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+            {"result": True, "comment": ""},
+            {"result": False, "comment": ""},
+            {"result": False, "comment": ""},
+        ]
+    )
+    with patch.dict(nftables.__salt__, {"nftables.check_chain": mock}):
+        ret.update(
+            {
+                "comment": "nftables salt chain is already"
+                " exist in filter table for ipv4"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:31]:
```suggestion
                "comment": "nftables salt chain is already exist in filter table for ipv4"
```

File: tests/pytests/unit/states/test_nftables.py [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+                {"result": False, "comment": ""},
+            ]
+        )
+        with patch.dict(nftables.__salt__, {"nftables.new_chain": mock}):
+            with patch.dict(nftables.__opts__, {"test": False}):
+                ret.update(
+                    {
+                        "changes": {"locale": "salt"},
+                        "comment": "nftables salt chain in filter"
+                        " table create success for ipv4",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:48]:
```suggestion
                        "comment": "nftables salt chain in filter table create success for ipv4",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+                    }
+                )
+                assert nftables.chain_present("salt") == ret
+
+                ret.update(
+                    {
+                        "changes": {},
+                        "comment": "Failed to create salt chain"
+                        " in filter table:  for ipv4",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:57]:
```suggestion
                        "comment": "Failed to create salt chain in filter table:  for ipv4",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+    """
+    Test to verify the chain is absent.
+    """
+    ret = {"name": "salt", "changes": {}, "result": True, "comment": ""}
+    mock = MagicMock(side_effect=[False, True])
+    with patch.dict(nftables.__salt__, {"nftables.check_chain": mock}):
+        ret.update(
+            {
+                "comment": "nftables salt chain is already absent"
+                " in filter table for ipv4"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:74]:
```suggestion
                "comment": "nftables salt chain is already absent in filter table for ipv4"
```

File: tests/pytests/unit/states/test_nftables.py [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+        )
+        assert nftables.chain_absent("salt") == ret
+
+        mock = MagicMock(return_value="")
+        with patch.dict(nftables.__salt__, {"nftables.flush": mock}):
+            ret.update(
+                {
+                    "result": False,
+                    "comment": "Failed to flush salt chain"
+                    " in filter table:  for ipv4",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:85]:
```suggestion
                    "comment": "Failed to flush salt chain in filter table:  for ipv4",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+                    {"comment": "nftables rule for salt" " already set (a) for ipv4"}
+                )
+                assert nftables.append("salt", table="", chain="") == ret
+
+                with patch.dict(nftables.__opts__, {"test": True}):
+                    ret.update(
+                        {
+                            "result": None,
+                            "comment": "nftables rule for salt needs"
+                            " to be set (a) for ipv4",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:119]:
```suggestion
                            "comment": "nftables rule for salt needs to be set (a) for ipv4",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 8]
```diff
@@ -0,0 +1,528 @@
...
+                            {"result": True, "comment": ""},
+                            {"result": False, "comment": ""},
+                        ]
+                    )
+                    with patch.dict(nftables.__salt__, {"nftables.append": mock}):
+                        ret.update(
+                            {
+                                "changes": {"locale": "salt"},
+                                "comment": "Set nftables rule for salt"
+                                " to: a for ipv4",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:136]:
```suggestion
                                "comment": "Set nftables rule for salt to: a for ipv4",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 9]
```diff
@@ -0,0 +1,528 @@
...
+                            }
+                        )
+                        assert nftables.append("salt", table="", chain="") == ret
+
+                        ret.update(
+                            {
+                                "changes": {},
+                                "comment": "Failed to set nftables"
+                                " rule for salt.\nAttempted rule was"
+                                " a for ipv4.\n",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:147]:
```suggestion
                                "comment": "Failed to set nftables rule for salt.\nAttempted rule was a for ipv4.\n",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 10]
```diff
@@ -0,0 +1,528 @@
...
+                    {"comment": "nftables rule for salt already" " set for ipv4 (a)"}
+                )
+                assert nftables.insert("salt", table="", chain="") == ret
+
+                with patch.dict(nftables.__opts__, {"test": True}):
+                    ret.update(
+                        {
+                            "result": None,
+                            "comment": "nftables rule for salt"
+                            " needs to be set for ipv4 (a)",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:182]:
```suggestion
                            "comment": "nftables rule for salt needs to be set for ipv4 (a)",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 11]
```diff
@@ -0,0 +1,528 @@
...
+                            {"result": True, "comment": ""},
+                            {"result": False, "comment": ""},
+                        ]
+                    )
+                    with patch.dict(nftables.__salt__, {"nftables.insert": mock}):
+                        ret.update(
+                            {
+                                "changes": {"locale": "salt"},
+                                "comment": "Set nftables rule for"
+                                " salt to: a for ipv4",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:199]:
```suggestion
                                "comment": "Set nftables rule for salt to: a for ipv4",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 12]
```diff
@@ -0,0 +1,528 @@
...
+                            nftables.insert("salt", table="", chain="", position="")
+                            == ret
+                        )
+
+                        ret.update(
+                            {
+                                "changes": {},
+                                "comment": "Failed to set nftables"
+                                " rule for salt.\nAttempted rule was"
+                                " a",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:213]:
```suggestion
                                "comment": "Failed to set nftables rule for salt.\nAttempted rule was a",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 13]
```diff
@@ -0,0 +1,528 @@
...
+                    {"result": True, "comment": ""},
+                    {"result": True, "comment": ""},
+                    {"result": True, "comment": ""},
+                ]
+            )
+            with patch.dict(nftables.__salt__, {"nftables.check": mock}):
+                ret.update(
+                    {
+                        "comment": "nftables rule for salt"
+                        " already absent for ipv4 (a)",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:245]:
```suggestion
                        "comment": "nftables rule for salt already absent for ipv4 (a)",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 14]
```diff
@@ -0,0 +1,528 @@
...
+                    }
+                )
+                assert nftables.delete("salt", table="", chain="") == ret
+
+                with patch.dict(nftables.__opts__, {"test": True}):
+                    ret.update(
+                        {
+                            "result": None,
+                            "comment": "nftables rule for salt needs"
+                            " to be deleted for ipv4 (a)",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:256]:
```suggestion
                            "comment": "nftables rule for salt needs to be deleted for ipv4 (a)",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 15]
```diff
@@ -0,0 +1,528 @@
...
+                            nftables.delete("salt", table="", chain="", position="")
+                            == ret
+                        )
+
+                        ret.update(
+                            {
+                                "result": False,
+                                "changes": {},
+                                "comment": "Failed to delete nftables"
+                                " rule for salt.\nAttempted rule was a",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:286]:
```suggestion
                                "comment": "Failed to delete nftables rule for salt.\nAttempted rule was a",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 16]
```diff
@@ -0,0 +1,528 @@
...
+                {"result": True, "comment": ""},
+                {"result": True, "comment": ""},
+            ]
+        )
+        with patch.dict(nftables.__salt__, {"nftables.check_table": mock}):
+            with patch.dict(nftables.__opts__, {"test": False}):
+                ret.update(
+                    {
+                        "comment": "Failed to flush table  in family"
+                        " ipv4, table does not exist.",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:315]:
```suggestion
                        "comment": "Failed to flush table  in family ipv4, table does not exist.",
```

File: tests/pytests/unit/states/test_nftables.py [hunk 17]
```diff
@@ -0,0 +1,528 @@
...
+                        {"result": False, "comment": ""},
+                        {"result": True, "comment": ""},
+                        {"result": True, "comment": ""},
+                    ]
+                )
+                with patch.dict(nftables.__salt__, {"nftables.check_chain": mock}):
+                    ret.update(
+                        {
+                            "comment": "Failed to flush chain  in table"
+                            "  in family ipv4, chain does not exist."
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:335]:
```suggestion
                            "comment": "Failed to flush chain  in table  in family ipv4, chain does not exist."
```

File: tests/pytests/unit/states/test_nftables.py [hunk 18]
```diff
@@ -0,0 +1,528 @@
...
+    ret = {"name": "salt", "changes": {}, "result": True, "comment": ""}
+
+    mock = MagicMock(return_value=[])
+    with patch.object(nftables, "_STATE_INTERNAL_KEYWORDS", mock):
+        mock = MagicMock(return_value="stack")
+        with patch.dict(nftables.__salt__, {"nftables.get_policy": mock}):
+            ret.update(
+                {
+                    "comment": "nftables default policy for chain"
+                    "  on table  for ipv4 already set to stack"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:382]:
```suggestion
                    "comment": "nftables default policy for chain  on table  for ipv4 already set to stack"
```

File: tests/pytests/unit/states/test_nftables.py [hunk 19]
```diff
@@ -0,0 +1,528 @@
...
+            )
+            assert (
+                nftables.set_policy("salt", table="", chain="", policy="stack") == ret
+            )
+
+            with patch.dict(nftables.__opts__, {"test": True}):
+                ret.update(
+                    {
+                        "comment": "nftables default policy for chain"
+                        "  on table  for ipv4 needs to be set to sal",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_nftables.py:393]:
```suggestion
                        "comment": "nftables default policy for chain  on table  for ipv4 needs to be set to sal",
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
Let black handle the string split/wrap

File: tests/pytests/unit/states/test_nftables.py (hunk 19 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_nftables.py:376]: line is too long for this case, black complains

File: tests/pytests/unit/states/test_nftables.py (hunk 18 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_nftables.py:365]: line it too long for black

File: tests/pytests/unit/states/test_nftables.py (hunk 19 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_nftables.py:376]: line is too long for black

Comment (s0undt3ch) [reply tests/pytests/unit/states/test_nftables.py:376]: then
```
"comment": (
    "nftables default policy for chain  on table  for "
    "ipv4 needs to be set to sal"
)
```
it's more readable

Comment (krionbsd) [reply tests/pytests/unit/states/test_nftables.py]: Fixed

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**